### PR TITLE
Added linux arm64 runners to terraform

### PR DESCRIPTION
**Pull request by DanilBaibak**
Added linux arm64 runners to terraform

### CODE REVIEW

**Review by jeanschmidt — APPROVED** [review]
Still need to validate scripts `terraform-aws-github-runner/modules/runners-instances/templates/*.sh` on canary, the rest LGTM

File: terraform-aws-github-runner/modules/runners-instances/variables.tf [hunk 1]
```diff
@@ -66,6 +71,21 @@ variable "ami_filter_linux" {
   }
 }
 
+variable "ami_filter_linux_arm64" {
+  description = "List of maps used to create the AMI filter for the action runner AMI."
+  type        = map(list(string))
+
+  default = {
+    name = ["amzn2-ami-hvm-2.*-arm64-gp2"]
```
Comment (DanilBaibak) [inline terraform-aws-github-runner/modules/runners-instances/variables.tf:79]: Is this the correct way to setup AMIs?

File: terraform-aws-github-runner/modules/runners-instances/templates/cloudwatch_config_linux_arm64.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ],
+    "metrics_collection_interval": 10
+  },
+  "swap": {
+    "measurement": [
+      "swap_used_percent"
+    ],
+    "metrics_collection_interval": 10
+  }
+}
```
Comment (DanilBaibak) [inline terraform-aws-github-runner/modules/runners-instances/templates/cloudwatch_config_linux_arm64.json:49]: Looks like the cloudwatch config is the same as for linux. Should we use the same?

File: terraform-aws-github-runner/modules/runner-binaries-syncer/main.tf [hunk 3]
```diff
@@ -9,8 +9,9 @@ terraform {
 }
 
 locals {
-  action_runner_distribution_object_key_linux   = "actions-runner-linux-x64.tar.gz"
-  action_runner_distribution_object_key_windows = "actions-runner-windows-x64.zip"
+  action_runner_distribution_object_key_linux       = "actions-runner-linux-x64.tar.gz"
+  action_runner_distribution_object_key_linux_arm64 = "actions-runner-linux-arm64-2.307.1.tar.gz"
```
Comment (DanilBaibak) [inline terraform-aws-github-runner/modules/runner-binaries-syncer/main.tf:13]: How to properly store actions runner?

**Comment by DanilBaibak** [comment]
Tested:

1. We have instances:
Screenshot 2023-08-10 at 14 05 24

2. We have GitHub runners:
Screenshot 2023-08-10 at 14 05 34

3. I was able to build linux aarch64 manywheel binaries:
Screenshot 2023-08-11 at 16 00 28

**Pull request merged**